src/CP/variableselection/failureBased.jl

```diff
@@ -0,0 +1,30 @@
+struct FailureBasedVariableSelection{TakeObjective} <: AbstractVariableSelection{TakeObjective} end
+
+FailureBasedVariableSelection(;take_objective=true) = FailureBasedVariableSelection{take_objective}()
+
+function (::FailureBasedVariableSelection{false})(cpmodel::CPModel)
+    selectedVar = nothing
+    minSize = Inf
+    for (k, x) in branchable_variables(cpmodel)
+        if x !== cpmodel.objective && !isbound(x) && length(x.domain) / cpmodel.statistics.infeasibleStatusPerVariable[id(x)] <= minSize
+            selectedVar = x
+            minSize = length(x.domain) / cpmodel.statistics.infeasibleStatusPerVariable[id(x)]
+        end
+    end
+    if isnothing(selectedVar) && !isbound(cpmodel.objective)
+        return cpmodel.objective
+    end
+    return selectedVar
+end
+
+function (::FailureBasedVariableSelection{true})(cpmodel::CPModel)
+    selectedVar = nothing
+    minSize = Inf
+    for (k, x) in branchable_variables(cpmodel)
+        if !isbound(x) && length(x.domain)/ cpmodel.statistics.infeasibleStatusPerVariable[id(x)] <= minSize
+            selectedVar = x
+            minSize = length(x.domain) / cpmodel.statistics.infeasibleStatusPerVariable[id(x)]
+        end
+    end
+    return selectedVar
+end
```

Comment:
Why do we need two almost similar functions ? 

`function (::FailureBasedVariableSelection{true})(cpmodel::CPModel)`

and 

`function (::FailureBasedVariableSelection{false})(cpmodel::CPModel)`

Reply:
We need to differentiate the cases where the objective variable is branchable and where it is not. 
Actually, we never undermined this part of the code that was made by ilan and Felix, maybe we could think of a more general way of doing it.

Reply:
Ok thanks. :-) Yes I could be refactored in the future in order to avoid duplicate code (I guess it happens more than once in the code).

But we can let it as it for now.